frame/composable-traits/src/bonded_finance.rs

```diff
@@ -0,0 +1,82 @@
+use frame_support::pallet_prelude::*;
+use scale_info::TypeInfo;
+use sp_runtime::{traits::Zero, ArithmeticError};
+
+use crate::math::SafeArithmetic;
+
+pub trait BondedFinance {
+	type AccountId;
+	type AssetId;
+	type Balance;
+	type BlockNumber;
+	type BondOfferId;
+
+	/// Create a new offer.
+	fn offer(
+		from: &Self::AccountId,
+		offer: BondOffer<Self::AssetId, Self::Balance, Self::BlockNumber>,
+	) -> Result<Self::BondOfferId, DispatchError>;
+
+	/// Bond for an offer.
+	fn bond(
+		offer: Self::BondOfferId,
+		from: &Self::AccountId,
+		contracts: Self::Balance,
+	) -> Result<Self::Balance, DispatchError>;
+}
+
+/// The Bond duration.
+#[derive(Clone, Encode, Decode, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub enum BondDuration<BlockNumber> {
+	/// Finite duration, liquidity is returned after a number of `blocks`.
+	Finite { blocks: BlockNumber },
```

Comment:
return_in : BlockNumbers 

in - in 1 hour, in 1 hours; return after like after 20211213T22:22:33

Reply:
Renamed :+1: